tests/auditors/client/test_client_must_not_use_global_wildcard_uri.py

```diff
@@ -0,0 +1,78 @@
+import pytest
+from unittest.mock import Mock
+
+from kcwarden.auditors.client.client_must_not_use_global_wildcard_uri import ClientMustNotUseGlobalWildcardURI
+
+
+class TestClientMustNotUseGlobalWildcardURI:
+    @pytest.fixture
+    def auditor(self, database, default_config):
+        auditor_instance = ClientMustNotUseGlobalWildcardURI(database, default_config)
+        auditor_instance._DB = Mock()
+        return auditor_instance
+
+    @pytest.mark.parametrize(
+        "is_oidc,has_standard_flow,has_implicit_flow,expected",
+        [
+            (True, True, False, True),  # OIDC with standard flow
+            (True, False, True, True),  # OIDC with implicit flow
+            (False, True, True, False),  # Non-OIDC should be excluded
+            (True, False, False, False),  # OIDC but no relevant flows
+        ],
+    )
+    def test_should_consider_client(
+        self, mock_client, auditor, is_oidc, has_standard_flow, has_implicit_flow, expected
+    ):
+        mock_client.is_oidc_client.return_value = is_oidc
+        mock_client.has_standard_flow_enabled.return_value = has_standard_flow
+        mock_client.has_implicit_flow_enabled.return_value = has_implicit_flow
+        assert auditor.should_consider_client(mock_client) == expected
+
+    # noinspection HttpUrlsUsage
+    @pytest.mark.parametrize(
+        "redirect_uri, should_alert",
+        [
+            ("*", True),  # Global wildcard
+            ("https://*", True),  # Arbitrary https link as wildcard
+            ("http://*", True),  # Arbitrary http link as wildcard
+            ("tel://*", True),  # Arbitrary other protocol link as wildcard
+            ("https://example.com/callback", False),  # No wildcard
```

Comment:
```suggestion
            ("*", True),  # Global wildcard
            ("https://*", True),  # Arbitrary https link as wildcard
            ("http://*", True),  # Arbitrary http link as wildcard
            ("tel://*", True),  # Arbitrary other protocol link as wildcard
            ("https://example.com/callback/*", False),  # Wildcard, but global (found by other auditor)
            ("https://example.com/callback", False),  # No wildcard
```

Reply:
I would suggest we also include a test with a non-global wildcard, which we would expect to return False, if I understand your intention correctly.